app/controllers/iterations_controller.rb

```diff
@@ -79,118 +49,101 @@ def update_all
-    newtask.iteration_id = params[:iteration_id]
-    newtask.description = params[:description]
-    newtask.save
+  # Edit a single task
+  def edit_task
+    task = Task.find(params[:task_id])
+    @iteration = Iteration.find(params[:iteration_id])
+    task.description = params[:description]
+    task.title = params[:title]
+    task.save
     redirect_to @iteration
   end
 
-  def aggregate_tasks_graph
-    if current_user.is_student?
-      if current_user.project.nil?
-        redirect_to init_user_path current_user
-      else
-        redirect_to project_path current_user.project
+  # Finalize the assignment of template to specific project with
+  # starting and ending date
+  def apply_to
+    project = Project.find params[:project_id]
+    newiter = Iteration.copy_assignment params[:id], project.id
+    if not newiter
+      respond_to do |format|
+        format.html {redirect_to iterations_path, notice: 'Iteration Template copy failed.'}
+      end
+    elsif not newiter.set_timestamp params[:start_time], params[:end_time]
+      newiter.destroy
+      respond_to do |format|
+        format.html { redirect_to iterations_path, notice: 'Failed. Please put in the start and end time according to the implied structure.' }
+      end
+    else
+      respond_to do |format|
+        format.html { redirect_to iterations_path, notice: 'Project was successfully assigned.' }
       end
-    end
-    @projects = Project.all
-    @progress = Iteration.task_progress
-    @tasks_iter = {}
-    @progress_hash = {}
-    @projects.each do |proj|
-      @progress_hash[proj.id] = Iteration.count_graph_status proj.id
-      iter = Iteration.where(project_id:proj.id)
-      tasks = Task.where(iteration_id: iter[0].id)
-      @tasks_iter[proj.id] = tasks
     end
   end
 
-  def apply_to
-    project = Project.find params[:project_id]
-    iteration = Iteration.find(params[:id])
-    newiter = Iteration.copy_assignment iteration, project.id
-    newiter.set_timestamp params[:start_time], params[:end_time]
-    redirect_to '/iterations'
+  # Apply the assignment to all projects selected
+  def apply_to_all
+    projects = params[:projects].split(' ')
+    projects.each do |project|
+      newiter = Iteration.copy_assignment params[:iteration_id], project
+      newiter.set_timestamp params[:start_time], params[:end_time]
+    end
+    respond_to do |format|
+      format.html { redirect_to iterations_path, notice: 'Projects was successfully assigned.' }
+    end
   end
 
+  # The Dashboard that is presenting all the iterations of all projects
   def dashboard
-    @projects = Project.all
+    if current_user.is_student?
+      redirect_to init_user_path current_user
+    end
     @progress = Iteration.task_progress
-    # temp value that will be class attribute when there are more than one class
-    @iteration_num = 4
+    @tasks_iter = Iteration.collect_current_tasks
   end
 
+  # Present and modify the template iteration graph
   def show_template
```

Comment:
You're using many instance variables in this method and the other one.  Is it possible to use a dictionary instead?  Something like {requestingTitles: .., planningTitles: ...}?